src/stable/build/commands/build/wasm_binary/compile.ts

```diff
@@ -1,14 +1,32 @@
 import { IOType } from 'child_process';
 
 import { execSyncPretty } from '#utils/exec_sync_pretty';
-import { AZLE_CARGO_TARGET_DIR } from '#utils/global_paths';
+import { AZLE_CARGO_TARGET_DIR, AZLE_ROOT } from '#utils/global_paths';
+
+/**
+ * Runs security checks (cargo audit and cargo deny check) before compilation.
+ * This helps ensure that dependencies don't have known security vulnerabilities
+ * or violate licensing policies.
+ */
+function runSecurityChecks(ioType: IOType): void {
```

Comment:
Please place this function below the compile function, as it's an internal helper function

Reply:
done